src/components/SearchAndFilter/SearchAndFilter.js

```diff
@@ -54,32 +60,107 @@ const SearchAndFilter = ({
       document.removeEventListener("keydown", keyDown);
     };
-  }, []);
+  }, [searchTerm]);
+
+  // Add passed chip to the searchData array
+  const addToSelected = (chip) => {
+    const currentSelected = [...searchData];
+    if (!currentSelected.includes(chip)) {
+      currentSelected.push(chip);
+      setSearchData(currentSelected);
+    }
+  };
+
+  // Remove passed chip from the searchData array
+  const removeFromSelected = (chip) => {
+    if (searchData.includes(chip)) {
+      const updatedSelected = searchData.filter(
+        (searchDataChip) => searchDataChip !== chip
+      );
+      setSearchData(updatedSelected);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    if (searchTerm !== "") {
+      addToSelected({ value: searchTerm });
+      setSearchTerm("");
+    }
+  };
+
+  // If the offsetTop is more than double height of a single chip, consider it
+  // overflowing
+  const updateFlowCount = function () {
+    const chips = searchContainerRef?.current?.querySelectorAll(".p-chip");
+    const overflowCount = overflowingChipsCount(chips, 1);
+    setOverflowSearchTermCounter(overflowCount);
+  };
+
+  useEffect(() => {
+    if (typeof ResizeObserver !== "undefined") {
+      const wrapperWidthObserver = new ResizeObserver(() => {
+        updateFlowCount();
+      });
+      const wrapper = searchContainerRef.current;
+      wrapperWidthObserver.observe(wrapper);
+    } else {
+      updateFlowCount();
+    }
+  }, [searchData]);
 
   return (
-    <div className="search-and-filter">
-      <SearchBox
-        autocomplete="off"
-        externallyControlled={externallyControlled}
-        placeholder="Search and filter"
-        onChange={(searchTerm) => searchOnChange(searchTerm)}
-        onClick={() => setFilterPanelHidden(false)}
-        onFocus={() => setFilterPanelHidden(false)}
-        value={searchTerm}
-      />
+    <div className="search-and-filter" ref={filterPanelRef}>
+      <div
+        className="search-and-filter__search-container"
+        aria-expanded={searchBoxExpanded}
+        ref={searchContainerRef}
+      >
+        {Object.values(searchData).map((chip) => {
+          return (
+            <Chip
+              lead={chip.lead}
+              value={chip.value}
+              key={`search-${chip.lead}+${chip.value}`}
+              onDismiss={() => removeFromSelected(chip)}
+              selected={true}
+            />
+          );
+        })}
+        <SearchBox
+          autocomplete="off"
+          externallyControlled={externallyControlled}
+          placeholder={searchData.length ? "Add filter" : "Search and filter"}
+          onChange={(searchTerm) => searchOnChange(searchTerm)}
+          onClick={() => setFilterPanelHidden(false)}
+          onFocus={() => setFilterPanelHidden(false)}
+          onSubmit={(e) => handleSubmit(e)}
+          value={searchTerm}
+        />
+        {overflowSearchTermCounter > 0 && (
+          <span
```

Comment:
Can you make this a button, please? This is currently not accessible

Reply:
Similar here, I'll add a `role="button"` which will make it accessible without requiring a ream of styling overrides